### PR TITLE
Fix `CPUParticles3D` using angle incorrectly when `ROTATE_Y` is set.

### LINKED ISSUE #89559: `CPUParticles3D` incorrectly applies the `angle` parameters when `particle_flag_rotate_y` is used.
### Tested versions

- v4.2.1.stable.official [b09f793f5]

### System information

Windows 10.0.19044 - Vulkan (Mobile) - dedicated Radeon RX Vega (Advanced Micro Devices, Inc.; 31.0.21910.5) - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Issue description

`CPUParticles3D` incorrectly applies the `angle_min` and `angle_max` properties when the `particle_flag_rotate_y` flag is used.

Specifically, the angle value is applied as angular velocity rather than a fixed value. This is contrary to the behavior when the `particle_flag_disable_z` flag is used. Additionally, `GPUParticles3D` correctly applies the angle value when an equivalent configuration is used.

Here is a video demonstrating the difference between `CPUParticles3D` and `GPUParticles3D`:

https://github.com/godotengine/godot/assets/2352020/556d8891-c993-4589-91b7-d22f066e32d0

And here are the properties used for each:

image image

### Steps to reproduce

1. Create a `CPUParticles3D` node with `particle_flag_rotate_y = true` and `angle_min` and `angle_max` set to an arbitrary non-zero range.
2. Observe that the particles spin extremely fast.

### Minimal reproduction project (MRP)

[cpuparticles3d-rotate_y-angle-repro.zip](https://github.com/godotengine/godot/files/14622805/cpuparticles3d-rotate_y-angle-repro.zip)

### PR DESCRIPTION
**Pull request by apples**
Resolves #89559.

### CODE REVIEW

**Comment by aXu-AP** [comment]
Seems to be working as intended 👍
rotation
Pictured: CpuParticles3D and GpuParticles3D with same settings.

File: scene/3d/cpu_particles_3d.cpp [hunk 1]
```diff
@@ -1124,7 +1124,7 @@ void CPUParticles3D::_particles_process(double p_delta) {
 			//turn particle by rotation in Y
 			if (particle_flags[PARTICLE_FLAG_ROTATE_Y]) {
 				Basis rot_y(Vector3(0, 1, 0), p.custom[0]);
-				p.transform.basis = p.transform.basis * rot_y;
+				p.transform.basis = rot_y;
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply scene/3d/cpu_particles_3d.cpp:1127]: Nevermind, we're only changing `basis`, so `origin` is untouched.

**Review by akien-mga — APPROVED** [review]
Looks good, and I confirmed this fixes the MRP.

File: scene/3d/cpu_particles_3d.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kleonc) [reply scene/3d/cpu_particles_3d.cpp:1127]: >The equivalent GPU particles code is this:

These look the same indeed. :+1:

**Review by kleonc — APPROVED** [review]

File: scene/3d/cpu_particles_3d.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (aXu-AP) [reply scene/3d/cpu_particles_3d.cpp:1127]: We are losing align with velocity though. But GpuParticles behavior seems to be similiar, it's one or another.

**Pull request merged**